.github/workflows/check-pr.yml

```diff
@@ -0,0 +1,24 @@
+name: Pull Request Validation
+
+on:
+  pull_request:
+    branches:
+      - trunk
+    types:
+      - labeled
+      - unlabeled
+      - opened
+      - reopened
+      - synchronize
+      - edited
```

Comment:
What's all this for? Could it just be PRs onto main?

Reply:
ignore the `trunk`, the `types` are all event types that can actually add a milestone:

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

Reply:
Why don't we need those for other actions?

Reply:
See the first Note in the link above:

> By default, a workflow only runs when a `pull_request` event's activity type is `opened`, `synchronize`, or `reopened`.

where `synchronize` means “A pull request's head branch was updated“.

That’s a set of events that’s perfect for running as a response to repo content changes.

But this workflow needs to run on metadata changes (labels and milestones) as well.

Reply:
Issues have a `milestoned`, `demilestoned` event. Do you know if `labelled` covers that for PRs?

Reply:
There seems to be a discrepancy between the workflow-focused events page and the webhook-focused event payload page. the former doesn’t mention the `[de]milestoned` events while the latter does:

- https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

  ![grafik](https://github.com/scverse/scanpy/assets/291575/63a9eef8-464e-4a6d-afec-119deb2fb1b5)


- https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request

  ![grafik](https://github.com/scverse/scanpy/assets/291575/180a812b-e0ce-4079-974f-69192f8b68d5)

Reply:
Here? https://github.com/SchemaStore/schemastore/blob/726fe2b26b20621a11e29f6f0de01c0979023c4c/src/schemas/json/github-workflow.json#L1352-L1377

I think the webhook payloads are more powerful than the GH workflow payloads: https://github.com/orgs/community/discussions/8674#discussioncomment-6219278

Reply:
Have you tested this somewhere?

I would like to know if this gets triggered when a milestone is added. Or if adding a `milestoned` would work.

> I think the official spec is what I linked above.

Which link? I thought these were all docs, and generally haven't had the best experience with the accuracy of github docs.

Reply:
> Or if adding a `milestoned` would work.

If so, it would be an undocumented feature and therefore not reliable.

As said: I’m pretty sure GitHub just shares the schema of workflow event payloads and webhook payloads. That doesn’t mean that the same list of action types can be used to trigger a workflow and a webhook.

Reply:
Tbh, I've found the documented features pretty unreliable too.

Reply:
Maybe submit a support request asking for clarification?

Reply:
OK, I checked: `milestoned` is triggered.

- [first run, no milestone](https://github.com/flying-sheep/playground/actions/runs/5313789551/jobs/9620154329#step:4:7):

  ![grafik](https://github.com/scverse/scanpy/assets/291575/ff32b6e3-50fd-46b4-bc18-5c9df75c0c55)

- [second run, milestoned event](https://github.com/flying-sheep/playground/actions/runs/5313797804/jobs/9620175670?pr=6)

  ![grafik](https://github.com/scverse/scanpy/assets/291575/6d5fe66a-8d35-4692-9ba3-95f3ac4942d0)

Reply:
haha we posted the same thing at the same time!